x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/prebuilt_rules_install_update_workflows.mdx

```diff
@@ -0,0 +1,190 @@
+# Rule Immutability / Customization
+
+## Test Plan for 2nd Milestone of Customizing prebuilt detection rules
+
+### Useful information
+
+Ticket: [https://github.com/elastic/security-team/issues/1974](https://github.com/elastic/security-team/issues/1974)
+
+**Assumptions**
+- The current test plan is only for Milestone 2 of the Rule Immutability/Customization feature to be released in 8.9. It does not pretend to cover any scenario for past or future milestones. Scenarios and flows are sensitive to change in future Milestones.
+- Below scenarios only apply to prebuilt rules.
+- Most of our users are on the 7.17.x version, that’s why the 8.x version is specified on scenarios, because this TestPlan is considering a minimum version of 8.x.
+- The rule Customization feature should be available to users on the Basic license and higher.
+
+
+### Scenarios
+
+### Notifications
+
+#### **Scenario: No callout messages are displayed when user does not have prebuilt rules installed**
+
+**GIVEN** user doesn't have any 8.x prebuilt rules installed  
+**AND** user is running a fresh instance  
+**WHEN** user navigates to the Rules Management Page  
+**THEN** no callouts message should be displayed
+
+#### **Scenario: No callout messages are displayed when there are no pending installs/updates**
+
+**GIVEN** user has the latest version of prebuilt rules `<prebuilt_rules_status>`  
+**WHEN** user navigates to the Rules Management Page  
+**THEN** no callout message is displayed for `<prebuilt_rules_status>` rules  
+
+*CASE 1: `<prebuilt_rules_status>` = installed*  
+*CASE 2: `<prebuilt_rules_status>` = updated*
+
```

Comment:
We should add two scenarios for the notifications:

**1. when a user navigates to the Add Elastic Rules page and all available rules have been installed, they should see a message saying exactly that, with a CTA to go back to Rules Management:**  
![image](https://github.com/elastic/kibana/assets/5354282/016c4181-72ae-4073-9a40-b26954b08451)



**2. when a user navigates to the Rule Updates tab and there are no available updates, they should see a message saying that All Elastic rules are up to date.**

![image](https://github.com/elastic/kibana/assets/5354282/72878103-69a4-4ed2-b990-91235a681e53)

Reply:
Synced with @vgomez-el: we will add these two scenarios.

Reply:
@jpdjere Regarding the second one:

> 2. when a user navigates to the Rule Updates tab and there are no available updates, they should see a message saying that All Elastic rules are up to date.

No available updates can be for 2 different reasons:

- all the installed prebuilt rules are up to date (each has the latest version available)
- no prebuilt rules are installed at all

In order to not handle these two in the UI (a proper message would need to be shown), let's hide the `Rule Updates` tab when there are no available updates.

Reply:
https://github.com/elastic/kibana/pull/160685#discussion_r1249006398

Reply:
Scenarios were updated accordingly and this PR implements the change: https://github.com/elastic/kibana/pull/161196